components/builder-scheduler/src/data_store.rs

```diff
@@ -262,6 +265,70 @@ impl DataStore {
                            r#"CREATE OR REPLACE FUNCTION set_project_state_ident_v1 (pid bigint, jid bigint, state text, ident text) RETURNS void AS $$
                                   UPDATE projects SET project_state=state, job_id=jid, project_ident=ident, updated_at=now() WHERE id=pid;
                            $$ LANGUAGE SQL VOLATILE"#)?;
+
+        // The job status table
+        migrator.migrate(
+            "scheduler",
+            r#"CREATE TABLE IF NOT EXISTS job_status (
+                                   id bigserial PRIMARY KEY,
+                                   job_id bigint,
+                                   owner_id bigint,
+                                   job_state text,
+                                   project_name text,
+                                   net_error_code int,
+                                   net_error_msg text,
+                                   build_started_at timestamptz,
+                                   build_finished_at timestamptz,
+                                   created_at timestamptz DEFAULT now()
+                            )"#,
+        )?;
+
+        // Insert a new job status
+        migrator.migrate("scheduler",
+                            r#"CREATE OR REPLACE FUNCTION insert_job_status_v1 (
+                               job_id bigint,
+                               owner_id bigint,
+                               job_state text,
+                               project_name text,
+                               net_error_code int,
+                               net_error_msg text,
+                               build_started_at timestamptz,
+                               build_finished_at timestamptz
+                           ) RETURNS SETOF job_status AS $$
+                                   BEGIN
+                                       RETURN QUERY INSERT INTO job_status (job_id, owner_id, job_state, project_name, net_error_code, net_error_msg, build_started_at, build_finished_at)
+                                           VALUES (job_id, owner_id, job_state, project_name, net_error_code, net_error_msg, build_started_at, build_finished_at)
+                                           RETURNING *;
+                                       RETURN;
+                                   END
+                               $$ LANGUAGE plpgsql VOLATILE
+                               "#)?;
+
+        migrator.migrate("scheduler",
+                        r#"CREATE OR REPLACE FUNCTION get_job_status_v1 (integer) RETURNS SETOF job_status AS $$
+                               BEGIN
+                                   RETURN QUERY SELECT * FROM job_status
+                                        ORDER BY created_at ASC
+                                        LIMIT $1;
+                                   RETURN;
+                               END
+                               $$ LANGUAGE plpgsql VOLATILE"#)?;
```

Comment:
Will tweak the function name to make it clearer.  The current usage is always 1, however like a number of other functions like the pending jobs ones, this is designed to eventually be able to handle more than just one status at a time.